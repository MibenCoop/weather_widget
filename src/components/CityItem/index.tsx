import React from "react";

import { City } from "../../interfaces";

export interface CityProps {
  city: City;
}

const CityItem: any = ({ city }: CityProps) => {
  // console.log("city", city);
  return <li style={{ textAlign: "left" }}>{city.city}</li>;
};

export default CityItem;
