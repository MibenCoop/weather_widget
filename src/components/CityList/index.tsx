import React from 'react';

import CityItem from '../CityItem';
import { City } from '../../interfaces';

export interface Props {
    cities: Array<any>;
}

const CityList: any = ({ cities }: Props) =>
    cities.map((item: City) => {
        // console.log("item", item);
        return <CityItem key={item.lat + item.lng} city={item} />;
    });

export default CityList;
