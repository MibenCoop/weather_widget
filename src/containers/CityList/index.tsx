import React, { Component, useState, useEffect } from 'react';
import { connect } from 'react-redux';

import { fetchCities } from '../../actions/cities';
import { increment } from '../../actions/counter';
import CitiesSelector from '../../selectors/citySelector';

import Spinner from '../../components/Spinner';
import CityList from '../../components/CityList';
import { SignupForm } from '../../components/SignupForm';
import Counter from '../../components/Counter';

import { City, CityListState, SelectedCities } from '../../interfaces';

interface Props {
    cities: Array<object>;
    biggestCities: Array<object>;
    counter: number;
    isFetching: boolean;
    fetchCities: Function;
}

interface State {
    playOrPause?: string;
    selectedCities?: SelectedCities;
    hasSelectedCities: boolean;
}

export function CityListContainer(props: Props) {
    const { fetchCities, biggestCities, cities, isFetching } = props;
    const [hasSelectedCities, setHasSelectedCities] = useState(false);
    const [selectedCities, setSelectedCities] = useState([]);
    useEffect(() => {
        const fetchCityList = () => {
            fetchCities();
        };
        return fetchCityList();
    }, []);

    const selectCities = ({ city, minTemperature, maxTemperature }: SelectedCities) => {
        const selectedCitiesBySelector = getCitiesByCelector(city, minTemperature, maxTemperature);
        setHasSelectedCities(true);
        setSelectedCities(selectedCitiesBySelector);
    };

    const getCitiesByCelector = (name: any, minTemperature: any, maxTemperature: any) => {
        const { cities }: any = props;
        if (name.length > 0) {
            return cities.filter((city: any) => city.name === name);
        } else if (minTemperature && maxTemperature) {
            return cities.filter(
                (city: any) => city.temperature <= maxTemperature && city.temperature >= minTemperature,
            );
        } else if (minTemperature) {
            return cities.filter((city: any) => city.temperature >= minTemperature);
        } else if (maxTemperature) {
            return cities.filter((city: any) => city.temperature <= maxTemperature);
        }
    };

    if (isFetching || !cities) return <Spinner />;
    return (
        <>
            <SignupForm selectCities={(values: SelectedCities) => selectCities(values)} />
            {hasSelectedCities ? (
                <>
                    <p>Selected cities</p>
                    <CityList cities={selectedCities} />
                </>
            ) : null}
            <p>Top ten biggest City in Russia</p>
            <CityList cities={biggestCities} />
            <p>All</p>
            <CityList cities={cities} />
        </>
    );
}

const mapStateToProps = ({ cities, counter }: CityListState) => {
    return {
        cities: cities.cities,
        isFetching: cities.isFetching,
        biggestCities: CitiesSelector(cities.cities),
        counter: counter,
    };
};

export default connect(mapStateToProps, {
    fetchCities: fetchCities,
    increment: increment,
})(CityListContainer);
