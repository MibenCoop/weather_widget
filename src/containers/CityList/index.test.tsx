import React from "react";
import { shallow } from "enzyme";

import CityList from "../../components/CityList";
import CityItem from "../../components/CityItem";

describe("City list container", () => {});
