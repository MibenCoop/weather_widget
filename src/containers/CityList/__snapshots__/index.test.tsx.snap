// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityList container should match snapshot 1`] = `
<Fragment>
  <Component
    selectCities={[Function]}
  />
  <p>
    Top ten biggest City in Russia
  </p>
  <CityList
    cities={
      Array [
        Object {
          "admin": "Moskva",
          "capital": "primary",
          "city": "Moscow",
          "country": "Russia",
          "iso2": "RU",
          "lat": "55.752222",
          "lng": "37.615556",
          "population": "10452000",
          "population_proper": "10452000",
        },
        Object {
          "admin": "Sankt-Peterburg",
          "capital": "admin",
          "city": "Saint Petersburg",
          "country": "Russia",
          "iso2": "RU",
          "lat": "59.894444",
          "lng": "30.264167",
          "population": "4553000",
          "population_proper": "3493212",
        },
        Object {
          "admin": "Novosibirskaya Oblast’",
          "capital": "admin",
          "city": "Novosibirsk",
          "country": "Russia",
          "iso2": "RU",
          "lat": "55.0415",
          "lng": "82.9346",
          "population": "1389000",
          "population_proper": "1037201",
        },
        Object {
          "admin": "Sverdlovskaya Oblast’",
          "capital": "admin",
          "city": "Yekaterinburg",
          "country": "Russia",
          "iso2": "RU",
          "lat": "56.8519",
          "lng": "60.6122",
          "population": "1313000",
          "population_proper": "1227976",
        },
        Object {
          "admin": "Nizhegorodskaya Oblast’",
          "capital": "admin",
          "city": "Nizhniy Novgorod",
          "country": "Russia",
          "iso2": "RU",
          "lat": "56.326944",
          "lng": "44.0075",
          "population": "1278000",
          "population_proper": "1214926",
        },
        Object {
          "admin": "Samarskaya Oblast’",
          "capital": "admin",
          "city": "Samara",
          "country": "Russia",
          "iso2": "RU",
          "lat": "53.200065",
          "lng": "50.150002",
          "population": "1137000",
          "population_proper": "856190",
        },
        Object {
          "admin": "Omskaya Oblast’",
          "capital": "admin",
          "city": "Omsk",
          "country": "Russia",
          "iso2": "RU",
          "lat": "55.0",
          "lng": "73.4",
          "population": "1135000",
          "population_proper": "1043403",
        },
        Object {
          "admin": "Tatarstan",
          "capital": "admin",
          "city": "Kazan’",
          "country": "Russia",
          "iso2": "RU",
          "lat": "55.78874",
          "lng": "49.122144",
          "population": "1115000",
          "population_proper": "912270",
        },
        Object {
          "admin": "Chelyabinskaya Oblast’",
          "capital": "admin",
          "city": "Chelyabinsk",
          "country": "Russia",
          "iso2": "RU",
          "lat": "55.154015",
          "lng": "61.42915",
          "population": "1091000",
          "population_proper": "946604",
        },
        Object {
          "admin": "Rostovskaya Oblast’",
          "capital": "admin",
          "city": "Rostov-na-Donu",
          "country": "Russia",
          "iso2": "RU",
          "lat": "47.23135",
          "lng": "39.723284",
          "population": "1052000",
          "population_proper": "1013134",
        },
      ]
    }
  />
  <p>
    All
  </p>
  <CityList
    cities={
      Array [
        Object {
          "admin": "Moskva",
          "capital": "primary",
          "city": "Moscow",
          "country": "Russia",
          "iso2": "RU",
          "lat": "55.752222",
          "lng": "37.615556",
          "population": "10452000",
          "population_proper": "10452000",
        },
        Object {
          "admin": "Sankt-Peterburg",
          "capital": "admin",
          "city": "Saint Petersburg",
          "country": "Russia",
          "iso2": "RU",
          "lat": "59.894444",
          "lng": "30.264167",
          "population": "4553000",
          "population_proper": "3493212",
        },
        Object {
          "admin": "Novosibirskaya Oblast’",
          "capital": "admin",
          "city": "Novosibirsk",
          "country": "Russia",
          "iso2": "RU",
          "lat": "55.0415",
          "lng": "82.9346",
          "population": "1389000",
          "population_proper": "1037201",
        },
        Object {
          "admin": "Sverdlovskaya Oblast’",
          "capital": "admin",
          "city": "Yekaterinburg",
          "country": "Russia",
          "iso2": "RU",
          "lat": "56.8519",
          "lng": "60.6122",
          "population": "1313000",
          "population_proper": "1227976",
        },
        Object {
          "admin": "Nizhegorodskaya Oblast’",
          "capital": "admin",
          "city": "Nizhniy Novgorod",
          "country": "Russia",
          "iso2": "RU",
          "lat": "56.326944",
          "lng": "44.0075",
          "population": "1278000",
          "population_proper": "1214926",
        },
        Object {
          "admin": "Samarskaya Oblast’",
          "capital": "admin",
          "city": "Samara",
          "country": "Russia",
          "iso2": "RU",
          "lat": "53.200065",
          "lng": "50.150002",
          "population": "1137000",
          "population_proper": "856190",
        },
        Object {
          "admin": "Omskaya Oblast’",
          "capital": "admin",
          "city": "Omsk",
          "country": "Russia",
          "iso2": "RU",
          "lat": "55.0",
          "lng": "73.4",
          "population": "1135000",
          "population_proper": "1043403",
        },
        Object {
          "admin": "Tatarstan",
          "capital": "admin",
          "city": "Kazan’",
          "country": "Russia",
          "iso2": "RU",
          "lat": "55.78874",
          "lng": "49.122144",
          "population": "1115000",
          "population_proper": "912270",
        },
        Object {
          "admin": "Chelyabinskaya Oblast’",
          "capital": "admin",
          "city": "Chelyabinsk",
          "country": "Russia",
          "iso2": "RU",
          "lat": "55.154015",
          "lng": "61.42915",
          "population": "1091000",
          "population_proper": "946604",
        },
        Object {
          "admin": "Rostovskaya Oblast’",
          "capital": "admin",
          "city": "Rostov-na-Donu",
          "country": "Russia",
          "iso2": "RU",
          "lat": "47.23135",
          "lng": "39.723284",
          "population": "1052000",
          "population_proper": "1013134",
        },
        Object {
          "admin": "Bashkortostan",
          "capital": "admin",
          "city": "Ufa",
          "country": "Russia",
          "iso2": "RU",
          "lat": "54.785167",
          "lng": "56.045621",
          "population": "1018000",
          "population_proper": "920756",
        },
        Object {
          "admin": "Permskiy Kray",
          "capital": "",
          "city": "Perm’",
          "country": "Russia",
          "iso2": "RU",
          "lat": "58.009282",
          "lng": "56.284063",
          "population": "997000",
          "population_proper": "851308",
        },
        Object {
          "admin": "Volgogradskaya Oblast’",
          "capital": "admin",
          "city": "Volgograd",
          "country": "Russia",
          "iso2": "RU",
          "lat": "48.71939",
          "lng": "44.501835",
          "population": "984000",
          "population_proper": "619655",
        },
        Object {
          "admin": "Krasnoyarskiy Kray",
          "capital": "admin",
          "city": "Krasnoyarsk",
          "country": "Russia",
          "iso2": "RU",
          "lat": "56.018393",
          "lng": "92.867174",
          "population": "925000",
          "population_proper": "302210",
        },
        Object {
          "admin": "Voronezhskaya Oblast’",
          "capital": "admin",
          "city": "Voronezh",
          "country": "Russia",
          "iso2": "RU",
          "lat": "51.672035",
          "lng": "39.184301",
          "population": "844000",
          "population_proper": "295469",
        },
        Object {
          "admin": "Saratovskaya Oblast’",
          "capital": "admin",
          "city": "Saratov",
          "country": "Russia",
          "iso2": "RU",
          "lat": "51.540556",
          "lng": "46.008611",
          "population": "843000",
          "population_proper": "786173",
        },
        Object {
          "admin": "Krasnodarskiy Kray",
          "capital": "admin",
          "city": "Krasnodar",
          "country": "Russia",
          "iso2": "RU",
          "lat": "45.044843",
          "lng": "38.976029",
          "population": "649851",
          "population_proper": "552532",
        },
        Object {
          "admin": "Ul’yanovskaya Oblast’",
          "capital": "admin",
          "city": "Ul’yanovsk",
          "country": "Russia",
          "iso2": "RU",
          "lat": "54.328241",
          "lng": "48.38657",
          "population": "640680",
          "population_proper": "502427",
        },
        Object {
          "admin": "Udmurtiya",
          "capital": "admin",
          "city": "Izhevsk",
          "country": "Russia",
          "iso2": "RU",
          "lat": "56.849756",
          "lng": "53.204479",
          "population": "631038",
          "population_proper": "591422",
        },
        Object {
          "admin": "Yaroslavskaya Oblast’",
          "capital": "admin",
          "city": "Yaroslavl’",
          "country": "Russia",
          "iso2": "RU",
          "lat": "57.61977",
          "lng": "39.85536",
          "population": "606730",
          "population_proper": "535578",
        },
        Object {
          "admin": "Altayskiy Kray",
          "capital": "admin",
          "city": "Barnaul",
          "country": "Russia",
          "iso2": "RU",
          "lat": "53.360556",
          "lng": "83.763611",
          "population": "599579",
          "population_proper": "539843",
        },
        Object {
          "admin": "Kemerovskaya Oblast’",
          "capital": "",
          "city": "Novokuznetsk",
          "country": "Russia",
          "iso2": "RU",
          "lat": "53.7557",
          "lng": "87.1099",
          "population": "539616",
          "population_proper": "521035",
        },
      ]
    }
  />
</Fragment>
`;
